test/e2e/curated_packages_test.go

```diff
@@ -4,37 +4,174 @@
 package e2e
 
 import (
+	"context"
 	"fmt"
+	"strconv"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/aws/eks-anywhere/internal/pkg/api"
 	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/test/framework"
 )
 
 const (
-	eksAnywherePackagesHelmChartName = "eks-anywhere-packages"
-	eksAnywherePackagesHelmUri       = "oci://public.ecr.aws/l0g8r8j6/eks-anywhere-packages"
-	eksAnywherePackagesHelmVersion   = "0.1.14-eks-a-v0.0.0-dev-build.3481"
-	eksAnywherePackagesBundleUri     = "oci://public.ecr.aws/l0g8r8j6/eks-anywhere-packages-bundles:v1-21-latest"
-
-	eksaPackageControllerHelmChartName = "eks-anywhere-packages"
-	eksaPackageControllerHelmURI       = "oci://public.ecr.aws/eks-anywhere/eks-anywhere-packages"
-	eksaPackageControllerHelmVersion   = "0.1.10-eks-a-10"
-	eksaPackageBundleBaseURI           = "oci://public.ecr.aws/eks-anywhere/eks-anywhere-packages-bundles"
+	EksaPackagesNamespace              = "eksa-packages"
+	EksaPackageControllerHelmChartName = "eks-anywhere-packages"
+	EksaPackagesSourceRegistry         = "public.ecr.aws/l0g8r8j6"
+	EksaPackageControllerHelmURI       = "oci://" + EksaPackagesSourceRegistry + "/eks-anywhere-packages"
+	EksaPackageControllerHelmVersion   = "0.1.14-eks-a-v0.0.0-dev-build.3624"
+	EksaPackageBundleURI               = "oci://" + EksaPackagesSourceRegistry + "/eks-anywhere-packages-bundles"
 )
 
-var (
-	eksaPackageControllerHelmValues = []string{}
-	eksAnywherePackagesHelmValues   = []string{"sourceRegistry=public.ecr.aws/l0g8r8j6"}
-)
+var EksaPackageControllerHelmValues = []string{"sourceRegistry=public.ecr.aws/l0g8r8j6"}
+
+type resourcePredicate func(string, error) bool
```

Comment:
Why does a predicate need to care about an error?

Reply:
A predicate could be `hasError(...)` for example.

Reply:
Given `resourcePredicate` is unexported meaning we have to provide the predicates from this package, and that we don't have any predicates that act on the error at all, I suggest we simplify by removing it.

Reply:
`NoErrorPredicate` does use the error input. We could indeed export `resourcePredicate` as well.

Reply:
https://github.com/aws/eks-anywhere/pull/2863/files#diff-e04741f3b0c5b687abeb91625437530ddf182779c1b1b5fa4df95b9c59717c5eR32

Reply:
I recall seeing that and thinking it should be general error handling for `Execute(...)`. Why do we need a separate predicate for it?

Reply:
It just allows to have test logic based on non-zero exit code and/or `stderr` from the forked process. If we use `kubectl wait --for` and it times out, we'll get a non-zero exit code with a stderr message which will be part of `error`.

Reply:
OK. Perhaps there's a broader change required. There's still something off here though - it sounds like a hard to use error return is dictating a package API.

Reply:
I suppose it being an `error` might not be ideal compared to having `stdout`, `stderr` and the return code.

Reply:
Two thoughts:

1. it seems poorly named, and should be documented if a more obvious name isn't forthcoming
2. if it's exported, it should probably
    a. have more documentation for future possible users 
    b. live in a shared utils package, for better discoverability